test/testutil/tests.c

```diff
@@ -145,6 +145,7 @@ void test_perror(const char *s)
 
 void test_note(const char *fmt, ...)
 {
+    test_flush_stdout();
```

Comment:
Is similar required in the other test output functions?
E.g. test_skip() below and test_error above?

Reply:
In theory probably.  However although TEST_NOTE is used in
a lot of test scripts, only the test script `test/evp_pkey_provided_test.c`
is affected due to the direct output of key data to `bio_out` without
explicit flushing.  There is no evidence for other test scripts to be affected.
But I consider this PR just to be a workaround for the issue,
maybe the underlying `tap_out` / `tap_err` BIO pair should handle this
somehow transparently.